src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/umbblockgridsortable.directive.js

```diff
@@ -0,0 +1,705 @@
+                    return vm;
+                };
+                containerEl.addEventListener('dragover', preventDragOver);
+
+                observer.observe(containerEl, {childList: true, subtree: false});
+            }
+            init();
+
+            function preventDragOver(e) {
+                e.preventDefault()
+            }
+
+            function setupItem(element) {
+
+                setupIgnorerElements(element);
+
+                element.draggable = true;
+                element.addEventListener('dragstart', handleDragStart);
+            }
+
+            function destroyItem(element) {
+
+                destroyIgnorerElements(element);
+
+                element.removeEventListener('dragstart', handleDragStart);
+            }
+
+            function setupIgnorerElements(element) {
+                config.ignorerSelector.split(',').forEach(function (criteria) {
+                    element.querySelectorAll(criteria.trim()).forEach(setupPreventEvent);
+                });
+            }
+            function destroyIgnorerElements(element) {
+                config.ignorerSelector.split(',').forEach(function (criteria) {
+                    element.querySelectorAll(criteria.trim()).forEach(destroyPreventEvent);
+                });
+            }
+            function setupPreventEvent(element) {
+                element.draggable = false
+            }
+            function destroyPreventEvent(element) {
+                element.removeAttribute('draggable');
+            }
+
+
+
+
+
+            let currentContainerElement = containerEl;
+            let currentContainerVM = vm;
+
+            let rqaId = null;
+            let currentItem = null;
+            let currentElement = null;
+            let currentDragElement = null;
+            let currentDragRect = null;
+            let dragX = 0;
+            let dragY = 0;
+
+            function handleDragStart(event) {
+
+                if(currentElement) {
+                    handleDragEnd();
+                }
+
+                event.stopPropagation();
+                event.dataTransfer.effectAllowed = "move";// copyMove when we enhance the drag with clipboard data.
+                event.dataTransfer.dropEffect = "none";// visual feedback when dropped.
+                
+                if(!scrollElement) {
+                    scrollElement = getParentScrollElement(containerEl, true);
+                }
+                
+                const element = event.target.closest(config.itemSelector);
+                
+                currentElement = element;
+                currentDragElement = config.draggableSelector ? currentElement.querySelector(config.draggableSelector) : currentElement;
+                currentDragRect = currentDragElement.getBoundingClientRect();
+                currentItem = vm.getItemOfElement(currentElement);
+                if(!currentItem) {
+                    console.error("Could not find item related to this element.")
```

Comment:
Should we show a warning to the user as well here?

Reply:
I'm not sure the user would want to see that. But I've added a return statement to ensure that the user doesn't drag something we cant find in the model.

Reply:
.. (Its not something I think is likely to happen.)